packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -604,6 +609,14 @@ export interface MuiPickersAdapter<TDate extends PickerValidDate, TLocale = any>
    * @returns {number} The number of the week of the given date.
    */
   getWeekNumber(value: TDate): number;
+  /**
+   * Get the number of the day of the week of the given date.
+   * The value is 1-based, 1 - first day of the week, 7 - last day of the week.
+   * @template TDate
+   * @param {TDate} value The given date.
+   * @returns {number} The number of the day of the week of the given date.
+   */
+  getDayOfWeek(value: TDate): number;
```

Comment:
I've for now [removed the reliance](https://github.com/mui/mui-x/pull/12173/commits/d030d7ed59ff5c4c02c6a25324852ec18e50755c) of this method because we can't reliably parse the value in a range field.
IMHO, the method makes sense, but we can remove it for now if you think it's better to limit the scope. 👌

Reply:
What is missing for your to be able to parse the value in a range field?

Reply:
Knowing which value from the array is the relevant one.
Even then, until such value is set, we'd need an alternative way of getting the day of the week from the section value. 🤔